airflow/executors/base_executor.py

```diff
@@ -617,6 +619,7 @@ def debug_dump(self):
             len(self.event_buffer),
             "\n\t".join(map(repr, self.event_buffer.items())),
         )
+        self.log.info("executor memory usgae (current: %d, peak: %d)", *tracemalloc.get_traced_memory())
```

Comment:
Would not that be an issue if `end` or `terminate` (or both) are overridden by executor classes? You cannot assume the executor classes call `tracemalloc.stop()`

Reply:
`start` is also regularly overridden by executors.

Can you add more (or any...) details to the PR description? What was the motivation for this feature? What problem are you trying to solve? This feels a bit like putting debugging logic into main.

Reply:
Yep, thanks for reminding me. I'll move them to `__init__` and `__del__` separately.

> This feels a bit like putting debugging logic into the main.

I just updated the description, and yes, the main reason for adding it is to ease the debugging process receiving sigurs2. or maybe let me change it to debug